src/Models/AbstractModel.php

```diff
@@ -24,4 +26,16 @@ public function getId() : int
     {
         return (int) $this->id;
     }
+
+    /**
+     * Execute an after softDelete
+     *
+     * @return void
+     */
+    public function beforeSoftDelete() : void
```

Comment:
and this has to be on afterSoftdelete not before

Reply:
We need to do it on beforeSoftDelete, because we need to delete first the relations of the entity so, when you have foreign keys on your database, dont throw error.